### ISSUE TITLE
Question about CIP labeling

### ISSUE THREAD

**Issue by shubbey** [SiMolecule/centres#3]
[ELIKOB.zip](https://github.com/SiMolecule/centres/files/9113909/ELIKOB.zip)

I've attached a molecule that has the two nitrogen centers labeled as "S" via Rule 3 (at least according to the jsmol integration: https://chemapps.stolaf.edu/jmol/jsmol/cip.htm

I'm wondering why these aren't assigned via Rule 1a, since the three branches aren't symmetric.  I'm assuming that your labeling is correct and I'm misunderstanding how the official assignment works.  Thanks

[Edit] Actually now I'm seeing inconsistent results.  I was under the impression that jsmol was built on the same code as Centres but when I drop that SDF into the above evaluator I get the aforementioned results.  When I run centres.jar I get a "no chiral" situation for this same file.  Any ideas what is correct here?  I am trying to implement these rules in-house and was under the impression that bridgehead nitrogens should be chiral contenders but by my logic these would be assigned via rule 1A and not rule 3 (Looking at the molecule itself makes me think that "no chiral" is correct since its basically planar but I didn't think dihedral angle was checked as part of the evaluation).

**Comment by johnmay** [comment]
Hmm, well that nitrogen is aromatic (if it's the one I think it is) so would be flat. Indeed it looks that way in the 3D file you provided. The other nitrogen would invert.

Easy to work with SMILES since the configurations are explicit- is this the one:
```
CCCCCN1C2=NC(=C([N@]2c2cc(ccc12)C(=O)OC)C(=O)c1ccccc1)c1ccccc1
CCCCCN1C2=NC(=C([N@@]2c2cc(ccc12)C(=O)OC)C(=O)c1ccccc1)c1ccccc1
```

**Comment by johnmay** [comment]
Paste in here: https://www.simolecule.com/cdkdepict/depict.html

![](https://www.simolecule.com/cdkdepict/depict/bow/svg?smi=CCCCCN1C2%3DNC(%3DC(%5BN%40%5D2c2cc(ccc12)C(%3DO)OC)C(%3DO)c1ccccc1)c1ccccc1&w=-1&h=-1&abbr=on&hdisp=bridgehead&showtitle=false&zoom=1.25&annotate=cip&r=0)
![](https://www.simolecule.com/cdkdepict/depict/bow/svg?smi=CCCCCN1C2%3DNC(%3DC(%5BN%40%40%5D2c2cc(ccc12)C(%3DO)OC)C(%3DO)c1ccccc1)c1ccccc1&w=-1&h=-1&abbr=on&hdisp=bridgehead&showtitle=false&zoom=1.25&annotate=cip&r=0)

**Comment by johnmay** [comment]
In-fact all Nitrogen are aromatic, just need to draw the fused benzene in an alternate Kekúle form.

**Comment by shubbey** [comment]
Thank you for the quick response.   But is there actually a chiral center here despite the labeling?  As noted, these are aromatic nitrogens and the ring system is flat. It seems to me that even tetrahedral nitrogens have such a low energy barrier that they would never be labeled chiral.  But all that said, if there is a standard I would like to be sticking to it.  So I'm just not sure here!

**Comment by johnmay** [comment]
Yes that’s what I’m saying. Sorry didn’t see your edit. 

There are two separate things, what things should be labelled and what is the label. I can’t comment too much on JMol since I work mainly on SMILES and 2D where everryrhing is nicely labelled. Coming from 3D if there is the slightest angle then it may be define. It sounds like maybe JSMol needs to be tweaked.

**Comment by johnmay** [comment]
I do have the bridge head code somewhere in CDK. As I said I mainly focused on SMILES and 2D in which case the input defines it. e.g. 

```
[N@]1(CC2)CC[C@H]2C(O)C1
```

![image](https://www.simolecule.com/cdkdepict/depict/bow/svg?smi=%5BN%40%5D1(CC2)CC%5BC%40H%5D2C(O)C1&w=-1&h=-1&abbr=on&hdisp=bridgehead&showtitle=false&zoom=1.25&annotate=none&r=0)

**Comment by shubbey** [comment]
Thanks again John.  I think I get it.  The issue here is more in the 3D-to-stereochemistry assignment than it is the CIP labeling (which in your examples is going off SMILES-provided chirality).  It seems to me that JMol has an issue because they are calling these planar nitrogens chiral (and also from what I can tell using the wrong CIP rule).   I suspect that N is a tricky case because it really will only be chiral in these bridgehead situations as you've just posted.  But I guess SMILES can call anything chiral as part of the designation.

**Comment by johnmay** [comment]
Correct. Interesting in the bridged example I posted, InChI does not considered that nitrogen chiral.

Here is the logic in the CDK: https://github.com/cdk/cdk/blob/9c95df9cf2d7fddbe393f805946ef57bd2faadbe/base/standard/src/main/java/org/openscience/cdk/stereo/Stereocenters.java#L523-L531

**Comment by shubbey** [comment]
Excellent!  That stereochemistry assignment info is very useful.

**Comment by iclkevin** [comment]
Sorry to bring this issue back to life. I have been playing with drawings of 3-Quinuclidinol, as in your post on July 14, 2022. Many users do draw this structure in 2D with protruding bonds for perspective figures, even though two of the bridges are degenerate.

<img width="247" height="197" alt="Image" src="https://github.com/user-attachments/assets/4e7c96ae-6329-438a-ad3d-132c7a3b9e98" />

Should these two bridgehead atoms be marked as stereocenters? Both Jmol (in 3D) and CENTRES (in 2D) marks them as S. Should CENTRES be designating stereo labels other than "ns" to these atoms? The closest examples I can find in the testing suite are VS128 and VS129, which designate the carbon bridgeheads as "ns".

I can also get this to work in CDK Depict, if I designate chirality in the SMILES string. This output differs from the one you show in 2022.

<img width="117" height="97" alt="Image" src="https://github.com/user-attachments/assets/c3361292-9ab3-446a-9a25-dad32397768b" />

Thank you for your time.

**Comment by johnmay** [comment]
Whether they should and whether they are by CIP are two different things. 

But I think yes in this case. There is a very simple test which I may have mentioned before, a stereo center is not a stereo center if flipping it allows you to find a mapping/bijection on to your self. If you have a substructure matcher which does stereochemistry checking it with SMARTS:

```
[N@]1(CC2)CC[C@H]2[C@H](O)C1 A
[N@]1(CC2)CC[C@@H]2[C@H](O)C1 B
[N@@]1(CC2)CC[C@H]2[C@H](O)C1 B
[N@@]1(CC2)CC[C@@H]2[C@H](O)C1 A
[N@]1(CC2)CC[C@H]2[C@@H](O)C1 C
[N@]1(CC2)CC[C@@H]2[C@@H](O)C1 D
[N@@]1(CC2)CC[C@H]2[C@@H](O)C1 D
[N@@]1(CC2)CC[C@@H]2[C@@H](O)C1 C
```

What you will see is that you only match when both are flipped - therefore both are required to be marked. 

Another good check is InChI which inspects automorphisms for stereochemistry. It doesn't support the nitrogen but this case is no different if it were a carbon and in that case it puts in 3 stereo indicators: `InChI=1S/C8H14O/c9-8-5-6-1-3-7(8)4-2-6/h6-9H,1-5H2/t6-,7-,8-/m0/s1` (``t6-,7-,8-``)


The fact they flip in pairs above indicates they are pseudo asymmetric buy they get assigned an upper case 'S/R'. Personally I think this is a mistake in CIP which was introduced at some point but I can't remember by who - a simpler case is seen here:

```
C[C@H]1OO[C@H](CCCCC)OO1
C[C@H]1OO[C@H](CCCC[C@H](Cl)O)OO1
```

The extra acyclic stereo centre changes whether it is r to S **despite** it is still pseudo asymmetric since obviously there is a rotate-able bond.

<img width="181" height="64" alt="Image" src="https://github.com/user-attachments/assets/577c3213-3084-485b-bfbb-1ac6e930788f" />

<img width="215" height="89" alt="Image" src="https://github.com/user-attachments/assets/1d560608-a922-4c17-a350-dfc616ccc4d8" />

I put oxygens in the ring here to show that that the acyclic center shouldn't really be affecting anything - and yet it does. Same the the other way around.

```
O[C@H]1CC[C@H](OCCCCC)CC1
O[C@H]1CC[C@H](OCCC[CH](Cl)O)CC1
O[C@H]1CC[C@H](OCCC[C@H](Cl)O)CC1
```

Very broken if you ask me - it makes name to structure much trickier as there will be older tools which perhaps didn't use this rule and has more correct descriptors (IMO). Another quirk is cubane is marked as non-stereo yet mathematically you *could* invert one of the stereocenters and it would be a different structure. Bob/Andrey are chemists are argued that "well that's not possible" but my counter point is you could add -CH2- in to the edges of cubane and the topology doesn't change - yet the *feature* of the current CIP rules persists and marks everything as *ns* even though you could isolate them.

Anyways if particularly one this one is being tricky (you didn't actually say what your issue was?) it could be an issue with the wedge detection/angles. Normally it is the middle bond which would be wedged so you might need a preprocessor/standardiser if users are drawing things badly. Sonja from the InChI trust would say "people need to learn (be taught how) to draw things correctly" I think it is more useful to try and detect and indicate/correct users when something is bad rather than reject it outright.

There is a section in the MDL/Symyx/Accerlys/BIOVIA manual about it where they call it the "triangle rule". See here - https://dscloud2.chemcart.com:9943/insight/help/content/pdf_files/bioviachemicalrepresentation.pdf.

**Comment by iclkevin** [comment]
Thanks, John. So if I understand correctly, the S designations on the bridgehead atoms are correct and intended via the CENTRES CIP implementation? And regardless of the capitalization, they are pseudo-asymmetric. When you discuss mathematical inversions, in this case the following image shows the bridges twisting, which would lead to the opposite (albeit impossible) configuration. Is this what you mean?

<img width="238" height="197" alt="Image" src="https://github.com/user-attachments/assets/f3af214c-defc-476b-96c9-ac812c16bc12" />

We use both the MDL and IUPAC specifications for 2D interpretation of stereo drawings. The user will choose which one they prefer. I don't think either paradigm is causing the problem. And of course, the 3D structure will always show designations on both bridgehead atoms, in consensus with the InChI output. The problem is there needs to be a consistent answer to why descriptors are appearing on these bridgehead atoms. It is either the CIP system is defined to do so, but chemistry-wise, it is nonsense; or these, in a chemistry sense, are stereocenters, or at least pseudo-asymmetric, and the designations are not just correct, but accurate. How would you best answer a user that says these bridgehead designations are wrong and they shouldn't be designated? Is there a section of the IUPAC Blue Book we should be referencing?

And of course, thank you for helping me on a Sunday!

**Comment by johnmay** [comment]
>  When you discuss mathematical inversions, in this case the following image shows the bridges twisting, which would lead to the opposite (albeit impossible) configuration. Is this what you mean?

Correct

To clarify you or your user doesn't think these should be labelled?

**Comment by iclkevin** [comment]
Thanks for asking :), and I have had time to digest this. I do have a user who has taken the position the bridgehead atoms should not be designated as stereocenters. I initially agreed as two of the bridges are degenerate and therefore the two bridgehead atoms have two substituents each that are constitutionally equivalent, and that would typically disqualify them from being considered chiral centers. However, in this case, they are spatially locked in such a way that if we only switch two of the substituents on one bridgehead atom, a reflection would not generate a superimposable configuration. Therefore, in a mathematical and chemistry sense, they are true chiral centers, and not psuedoasymmetric, and given capital designations.

I do see you as the expert for this topic, and it is your position that I will defend (I think my position matters the least, less than the user's and less than yours). In your previous response, you state "The fact they flip in pairs above indicates they are pseudo asymmetric but they get assigned an upper case 'S/R'. Personally I think this is a mistake in CIP which was introduced at some point but I can't remember by who".

"A pseudo-asymmetric centre is found in a meso molecule where a plane of symmetry runs through a stereogenic centre and that stereogenic centre’s two substituents are constitutionally the same (same atoms connected by the same type of bonds) but configurationally different (spatially different)." https://chemistryschool.net/pseudo-asymmetric-centres/

However, there is no plane of symmetry through the bridgehead atoms.

<img width="635" height="500" alt="Image" src="https://github.com/user-attachments/assets/611a77b5-3bfd-4aa8-bd3e-5b45ec0df58e" />

So I am not sure what the true answer is here. Can you please clarify what your position is and if you agree with my conclusion?

**Comment by johnmay** [comment]
Pseudo-asymmetry is more broad than just meso compounds although they are the common case, see this better definition: https://goldbook.iupac.org/terms/view/P04921. If you remove the hydroxy stereocenter your should be able to confirm you (correctly) label them as pseudo asymmetric as well: 

```
[N@]1(CC2)CC[C@H]2C(=O)C1 (r,r)-
[N@]1(CC2)CC[C@@H]2C(=O)C1 (s,s)- 
```

As I said it is a technical 'bug' in CIP when you have another stereo-center it causes them to be non-pseudo asymmetric.

Anyways that is an aside, the argument is that because the (r,r)- molecule (ring system) is spacially locked/sterically hindered it should not have a stereo-descriptor? This sounds reasonable but completely independent of the CIP labelling. CIP is not designed or meant to do that, it is meant to name isomers not tell you if they can/can't exist.

You could write some logic to block this case but you just kick the problem down the road. 
The trouble is beyond simple cases knowing something is locker/hindered is very tricky.

Consider the following:

<img width="440" height="357" alt="Image" src="https://github.com/user-attachments/assets/b9293bf7-94f5-4a1c-875e-1fc8c2e389be" />

When n=1 we have [3-Quinuclidone](https://en.wikipedia.org/wiki/3-Quinuclidone) and the atoms are spatially locked, the other isomer is 'theoretical' only. At what *n* are the atoms no longer spatially locked? At some point you have what IUPAC calls in-out isomerism: https://goldbook.iupac.org/terms/view/I03055. 

Knowing where the line is drawn is fuzzy and depends on many factors. Similar to other problems like asking about many colors are there in the rainbow? When length does a polypeptide become a protein?


In summary I think it is possibly reasonable to add an algorithm which could block these from being labelled/assigned CIP descriptors and some other cases. It is however not the job of CIP to do this.

**Comment by iclkevin** [comment]
Ok, thanks John. So just to confirm. In 3-Quinuclidone, the bridgehead atoms are both pseudoasymmetric (s,s). They are labelled with capital letters due to a flaw in the design of CIP rules. Therefore, the S descriptors are correct, from a CIP perspective?

I believe they should be labelled. Just because the "right handed" configuration does not exist does not mean that the "left handed" configuration is not still "left handed".

I appreciate all of your guidance on this topic.

**Comment by johnmay** [comment]
> Ok, thanks John. So just to confirm. In 3-Quinuclidone, the bridgehead atoms are both pseudoasymmetric (s,s). They are labelled with capital letters due to a flaw in the design of CIP rules. Therefore, the S descriptors are correct, from a CIP perspective?

No - **3-quinuclidone** should be (s,s)- by CIP as it currently is, that one is fine. It is your original example **(R or S)-3-hydroxyquinuclidine** causes one of the bridge heads to be capital R/S which is wrong since that extra stereo-center doesn't change anything. This is orthogonal to the issue though.

> I believe they should be labelled. Just because the "right handed" configuration does not exist does not mean that the "left handed" configuration is not still "left handed".

Yep exactly

**Comment by iclkevin** [comment]
My apologies, I meant R-3-Quinuclidinol, I copied the wrong name. Both bridgehead configurations are capital, as output by CENTRES, not just one. But as you say, this is technical 'bug' in CIP, and therefore the capital designations are correct from a CIP perspective? Or is this a bug in the implemention of CIP rules?

**Comment by johnmay** [comment]
Correct as per the CIP rules and not a bug in implementation

**Comment by iclkevin** [comment]
John, you are the best!
